### PR TITLE
chore: run `yarn spec` on npm version

### PR DESCRIPTION
**Pull request by haxxmaxx**
Based on @cbt1's changes for mekko, `yarn spec` is added to run at the same time as version. Build  runs before to make sure the build step wont fail on Circle. Also extending the instructions a bit.

### CODE REVIEW

File: .github/CONTRIBUTION.md [hunk 1]
```diff
@@ -19,11 +19,25 @@ Run lint with:
...
+- Bumping the package version based on your commits
+- Updating the API specification
+- Creating a new commit with the changed files
+- Creating a tag with the new version
+- Pushing the release commit and tag to master
+
+### Step-By-Step
+
+1. Check out master and run `git pull`.
+1. Run `git clean -dfx && yarn` to make sure depenencies are up-to-date.
```
Comment (cbt1) [inline .github/CONTRIBUTION.md:35]: Looks like the numbers are not getting incremented

**Review by cbt1 — APPROVED** [review]
Fix the number increment and you are good to go

File: .github/CONTRIBUTION.md [hunk 2]
```diff
@@ -19,11 +19,25 @@ Run lint with:
...
+1. Run `npm version [major | minor | patch] -m "chore(release): v%s"`. Use semver string based on conventional commits since last release. Ex: `npm version patch -m "chore(release): v%s"`.
+1. Run `git push && git push --tags` to push commit and tag.
+1. Make sure all checks pass, then Circle CI automatically publishes to NPM.
+
+### On version command failure
```
Comment (LiKang6688) [inline .github/CONTRIBUTION.md:40]: I do not understand this section

File: .github/CONTRIBUTION.md (hunk 1 above)
Comment (haxxmaxx) [reply .github/CONTRIBUTION.md:35]: No need to, that is a way of creating a numbered list with markdown. It is easier if you would add a step inbetween already existing steps for example.

File: .github/CONTRIBUTION.md (hunk 2 above)
Comment (cbt1) [reply .github/CONTRIBUTION.md:40]: Actually good catch. This section doesn't make sense in the context of failed npm version command. You should not end up in a scenario where a tag is created if `preversion` or `version` scripts fails.

But it should perhaps be updated to include instructions on how to revert `npm version` if one is for some reason not happy with the end result.

File: .github/CONTRIBUTION.md (hunk 1 above)
Comment (cbt1) [reply .github/CONTRIBUTION.md:35]: Cool! Did not know 👍

File: .github/CONTRIBUTION.md (hunk 2 above)
Comment (haxxmaxx) [reply .github/CONTRIBUTION.md:40]: For example, if a version number already exists on your machine, npm version will fail. Then you have to remove it and try again

Comment (LiKang6688) [reply .github/CONTRIBUTION.md:40]: I see. Thanks. Could you also add those explanations into the docs?

Comment (haxxmaxx) [reply .github/CONTRIBUTION.md:40]: OK I can clarify a bit.

Comment (haxxmaxx) [reply .github/CONTRIBUTION.md:40]: fixed

**Review by LiKang6688 — APPROVED** [review]

**Pull request merged**